src/topology/homotopy/fundamental_groupoid.lean

```diff
@@ -291,3 +295,45 @@ instance : category_theory.groupoid (fundamental_groupoid X) :=
   comp_inv' := λ x y f, quotient.induction_on f
     (λ a, show ⟦a.trans a.symm⟧ = ⟦path.refl x⟧,
           from quotient.sound ⟨(path.homotopy.refl_trans_symm a).symm⟩) }
+
+lemma comp_eq (x y z : fundamental_groupoid X) (p : x ⟶ y) (q : y ⟶ z) :
+  p ≫ q = quotient.lift₂ (λ (l₁ : path x y) (l₂ : path y z), ⟦l₁.trans l₂⟧) begin
```

Comment:
Are you sure you don't want `quotient.map₂` here instead?

Reply:
Yeah, I think it does make the code a bit shorter. There's a bit of API missing for `quotient.map₂` which is why I was using `lift` originally.